### ISSUE TITLE
[docdb] The last op in the WAL with id has a term (7) that is greater than the latest recorded term, which is 6

### ISSUE THREAD

**Issue by pilshchikov** [yugabyte/yugabyte-db#14489]
Jira Link: [DB-3883](https://yugabyte.atlassian.net/browse/DB-3883)
### Description

Env:
- run 2 clusters with RF=3 3 nodes c5.xlarge
- setup bi-directional xcluster replication with platform UI (2 replication source <-> sink)
- create 1 table with 11 columns with different datatypes (column names k primary, v1-v11 with different datatypes)
- cluster/platform version:  2.17.1.0-b46 

Steps:
1. Load data for some time (2-15 min) in table on both clusters - with different primary key parts: for example in cluster 1 `k = "[UUID1]:key:1"` in cluster 2 `k = "[UUID2]:key:1"`. Different keys but in one table
2. In same moment in background do nemesis actions on both clusters: Restart VM, Slow network between nodes in cluster, Start/stop VM, start/stop master/tserver process
3. Check that information is replicated
4. Check that cluster has no FATAL logs

Found FATAL log on "source" universe (there is no source or sink, because it bi-directional, but still need somehow to tag this universes):
```
Log file created at: 2022/10/16 23:39:58
Current UTC time: 2022/10/16 23:39:58
Running on machine: ip-172-151-27-150.us-west-2.compute.internal
Application fingerprint: version 2.17.1.0 build 46 revision 7eb865180299dadb948b0ae9de7b4e655ce4b5b7 build_type RELEASE built at 16 Oct 2022 07:13:52 UTC
Node information: { hostname: 'ip-172-151-27-150.us-west-2.compute.internal', rpc_ip: '172.151.27.150', webserver_ip: '172.151.27.150', uuid: 'd3dfd867ebdc401aaeac2549aa5f8765' }
Running duration (h:mm:ss): 0:11:56
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
F1016 23:39:58.592168  1426 replica_state.cc:113] Check failed: last_id_in_wal.term() <= GetCurrentTermUnlocked() (7 vs. 6) T abe4da0a2d914b9ebe59d9a50ade196f P d3dfd867ebdc401aaeac2549aa5f8765 [term 6 FOLLOWER]: The last op in the WAL with id 7.1259245 has a term (7) that is greater than the latest recorded term, which is 6

```

All logs can be found here: http://stress.dev.yugabyte.com/stress_test/88cc522a-1345-454f-85c5-badef1c2749c
Download all logs: Attachments -> Download all (icon)

When FATAL log is happen there is no nemesis was executed on any clsuter

**Comment by qvad** [comment]
Also reproduced on non xcluster configuration during our tablet splitting tests with large keys.
```
Log file created at: 2022/11/30 21:03:36
Current UTC time: 2022/11/30 21:03:36
Running on machine: ip-172-151-16-182.us-west-2.compute.internal
Application fingerprint: version 2.17.1.0 build 271 revision 1378be4217af5c918f11b3914417baca3c9f3e7b build_type RELEASE built at 30 Nov 2022 17:15:08 UTC
Node information: { hostname: 'ip-172-151-16-182.us-west-2.compute.internal', rpc_ip: '172.151.16.182', webserver_ip: '172.151.16.182', uuid: 'c416402997ab4afaba03a98c9c6f64e6' }
Running duration (h:mm:ss): 0:07:56
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
F1130 21:03:36.649976 24197 replica_state.cc:112] Check failed: last_id_in_wal.term() <= GetCurrentTermUnlocked() (13 vs. 12) T 0a8f8c961828419ebfdc3a4a66a86030 P c416402997ab4afaba03a98c9c6f64e6 [term 12 LEARNER]: The last op in the WAL with id 13.53605 has a term (13) that is greater than the latest recorded term, which is 12
```

**Comment by Huqicheng** [comment]
This issue should be introduced by the commit for copying in-memory retryable requests structure: https://github.com/yugabyte/yugabyte-db/commit/341db1b4852b17ff42bf5d8da7db2e73af7edf81

In this commit, we load consensus metadata before BootstrapTablet (where current_term can be updated).

**Comment by bmatican** [comment]
@Huqicheng can you clarify if this will cause a crash loop? also, is there a way to quickly disable the copying of in-memory retryable requests?

**Comment by Huqicheng** [comment]
@bmatican It will not cause crash loop because we will re-load consensus meta data at restart and the term will be up-to-date.

**Comment by bmatican** [comment]
Got it! Ok, so then that's a bit less critical and we could potentially fix it in 2.16.1 and not block 2.16.0 cc @rthallamko3

**Comment by shamanthchandra-yb** [comment]
This issue is also seen in our latest packed columns toggle OFF and ON + tserver restart, testcase.
[http://stress.dev.yugabyte.com/stress_test/59f665fc-6802-4bcd-ac49-cd72cdd0f0c3](http://stress.dev.yugabyte.com/stress_test/59f665fc-6802-4bcd-ac49-cd72cdd0f0c3)

cc: @kripasreenivasan
